CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing to code
+
+Always branch off branch `main` before commiting changes, then push to remote and open a PR into `main`.
+A developer will then approve the PR and release.
+
+## Before making a pull request
+
+### Commit history
+
+Commit messages should reference a specific issue wherever possible.
+
+### Linting and formatting
+
+We use [ruff](https://docs.astral.sh/ruff/) to lint and format this repository. The ruff settings are defined in [pyproject.toml](./pyproject.toml).
+You can check for linting errors by running `ruff check .`. This should be run within the ONE root directory for the pyproject configuration to be used.
+The linter also checks for docstring formatting.
+
+### Testing and coverage
+
+You should ensure all tests are running locally.  We run tests using python unittest: `python -m unittest discover`. This should be run within the ONE root directory.  Coverage should also be checked as we keep coverage at 100%. Testing with coverage can be run with the following: `coverage run --omit=one/tests/* -m unittest discover`.
+
+## Releasing (maintainers only)
+
+Before merging to main, the CI checks that all unit tests pass and that coverage does not decrease.  Merges should not occur until all checks pass.
+The Github testing workflow can be in [.github/workflows/main.yaml](.github/workflows/main.yaml) and is testing against Python version 3.10 and 3.12 on Ubuntu and Windows.
+The divergent changes to the `docs` branch should be pulled into your release branch as the [.github/workflows/docs.yaml](.github/workflows/docs.yaml) workflow is triggered by both pushes to main and docs.
+
+### Release notes and version
+On a release or feature branch, update the [CHANGELOG.md](./CHANGELOG.md) file with a terse list of changes for the release, oriented for users.  We organize changes by 'added' - new functions/modules/packages; 'modified' - changes to current functions and modules; 'removed' - deleted functions/modules/packages.  A short one or two sentence description below the version heading should describe the principle changes.  For example:
+
+```markdown
+## [Latest](https://github.com/int-brain-lab/ONE/commits/main) [X.X.X]
```

Comment:
We could put the link here :https://keepachangelog.com/en/1.1.0/

Reply:
Sure! I was unaware of this site